python/ml4ir/base/tests/test_stat_analysis_calculation.py

```diff
@@ -0,0 +1,188 @@
+
+    def running_power_analysis_test(self, metric_list, group_key, group_metric_running_variance_params):
+        # performing power analysis
+        group_metrics_stat_sig = run_power_analysis(metric_list,
+                                                    group_key,
+                                                    group_metric_running_variance_params,
+                                                    0.9,
+                                                    0.1)
+        for row in group_metrics_stat_sig.iterrows():
+            for m in metric_list:
+                stat_check = "is_" + m + "_lift_stat_sig"
+                group_name = row[1][str(group_key)]
+                if m == "metric1" and group_name == 'A':
+                    assert row[1][stat_check] == True
+                else:
+                    assert row[1][stat_check] == False
+
+    def test_1(self):
+        n = 100
+        batch_size = 10
+        batch_a = np.random.randn(n) + 1
+        batch_b = np.random.randn(n)
+        self.t_test_calculation(batch_a, batch_b, batch_size)
+
+    def test_2(self):
+        n = 10
+        batch_size = 3
+        batch_a = np.random.randn(n) + 20
+        batch_b = np.random.randn(n)
+        self.t_test_calculation(batch_a, batch_b, batch_size)
+
+    def test_3(self):
+        n = 50
+        batch_size = 5
+        batch_a = np.random.randn(n)
+        batch_b = np.array(batch_a)
+        batch_b[0] += 1
+        self.t_test_calculation(batch_a, batch_b, batch_size)
+
+    def test_4(self):
+        n = 50
+        batch_size = 5
+        batch_a = np.random.randn(n)
+        batch_b = np.array(batch_a)
+        batch_b[0] += 1
+        self.t_test_calculation(batch_a, batch_b, batch_size)
+
+    def test_5(self):
+        n = 100
+        batch_size = 5
+        batch_a = np.random.randn(n)
+        batch_b = np.random.randn(n)
+        self.t_test_calculation(batch_a, batch_b, batch_size)
+
+    def test_6(self):
+        n = 100
+        batch_size = 5
+        batch_a = np.random.exponential(scale=0.1, size=n)
+        batch_b = np.random.exponential(scale=0.5, size=n)
+        self.t_test_calculation(batch_a, batch_b, batch_size)
+
+    def test_7(self):
+        n = 10
+        batch_size = 2
+        batch_a = np.random.exponential(scale=1.0, size=n)
+        batch_b = np.random.exponential(scale=0.2, size=n)
+        self.t_test_calculation(batch_a, batch_b, batch_size)
+
+    def test_8(self):
+        n = 50
+        batch_size = 10
+        batch_a = np.random.exponential(scale=0.1, size=n)
+        batch_b = np.random.exponential(scale=0.1, size=n)
+        self.t_test_calculation(batch_a, batch_b, batch_size)
+
+    def test_9(self):
+        n = 50
+        batch_size = 5
+        batch_a = np.random.exponential(scale=0.1, size=n)
+        batch_b = batch_a + 0.001
+        self.t_test_calculation(batch_a, batch_b, batch_size)
```

Comment:
@mohazahran Thanks for the unit tests, can you also rename them to something appropriate? Add a short docstring about what you are trying to test for reference.

Reply:
ping

You can maybe make them `self.subTest("...")`